src/utils/FileBuffer.cpp

```diff
@@ -0,0 +1,285 @@
+#include "FileBuffer.hpp"
+
+#include <cstdlib>
+#include <ctime>
+#include <libftpp/expected.hpp>
+#include <libftpp/string.hpp>
+#include <utils/IBuffer.hpp>
+
+#include <cstddef>
+#include <cstdio>
+#include <fstream>
+#include <ios>
+#include <iostream>
+#include <string>
+#include <sys/stat.h>
+
+/* ************************************************************************** */
+// INIT
+
+const char* const FileBuffer::errOpen = "FbException: open failed";
+const char* const FileBuffer::errSeek = "FbException: seek failed";
+const char* const FileBuffer::errFileEmpty = "FbException: file is empty";
+const char* const FileBuffer::errFileNotOpened =
+  "FbException: file is not opened";
+const char* const FileBuffer::errEof = "FbException: tried to get end of file";
+const char* const FileBuffer::errRead = "FbException: read failed";
+const char* const FileBuffer::errWrite = "FbException: write failed";
+
+/* ************************************************************************** */
+// PUBLIC
+
+FileBuffer::FileBuffer()
+  : _size(0)
+{
+}
+
+FileBuffer::~FileBuffer()
+{
+  if (_fs.is_open()) {
+    _fs.close();
+    (void)std::remove(_fileName.c_str());
+  }
```

Comment:
The file should always be tried to be removed, otherwise it's possible on exceptions that it will not be.
```suggestion
  if (_fs.is_open()) {
    _fs.close();
  }
  if (!_fileName.empty()) {
    (void)std::remove(_fileName.c_str());
  }
```

```
  tmp._fs.close();

  // Replace current file with tmp file
  _fs.close();
  (void)std::remove(_fileName.c_str());
  _fileName = tmp._fileName;
```
In this example from further below, if the assignment throws, `tmp._fs` is closed already and its file will not be removed.